api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java

```diff
@@ -0,0 +1,237 @@
+
+        assertThat(cloudRegionApiService.delete(region.getId())).isEqualTo(region);
+    }
+
+    @Test(expected = AccessDeniedException.class)
+    @WithMockUser(roles = SIMPLE_USER_ROLE)
+    public void shouldDeleteAbstractCloudRegionForNotAdmin() {
+        doReturn(region).when(cloudRegionManager).delete(region.getId());
+
+        cloudRegionApiService.delete(region.getId());
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldLoadAllAvailableCloudsForAdmin() {
+        doReturn(availableCloudsList).when(cloudRegionManager).loadAllAvailable(CloudProvider.AWS);
+
+        assertThat(cloudRegionApiService.loadAllAvailable(CloudProvider.AWS)).isEqualTo(availableCloudsList);
+    }
+
+    @Test(expected = AccessDeniedException.class)
+    @WithMockUser(roles = SIMPLE_USER_ROLE)
+    public void shouldNotLoadAllAvailableCloudsForNotAdmin() {
+        doReturn(availableCloudsList).when(cloudRegionManager).loadAllAvailable(CloudProvider.AWS);
+
+        cloudRegionApiService.loadAllAvailable(CloudProvider.AWS);
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnAllCloudRegionsForAdmin() {
+        doReturn(clouds).when(cloudRegionManager).loadAll();
+
+        List<? extends AbstractCloudRegion> abstractCloudRegions = cloudRegionApiService.loadAll();
+
+        assertThat(abstractCloudRegions).isEqualTo(clouds);
+        assertThat(abstractCloudRegions.get(0).getId()).isEqualTo(clouds.get(0).getId());
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER_ROLE)
+    public void shouldReturnAllCloudRegionsWhenPermissionIsGranted() {
+        initAclEntity(region,
+                Collections.singletonList(new UserPermission(SIMPLE_USER_ROLE, AclPermission.READ.getMask())));
+        doReturn(clouds).when(cloudRegionManager).loadAll();
+
+        List<? extends AbstractCloudRegion> abstractCloudRegions = cloudRegionApiService.loadAll();
+
+        System.out.println("abstractCloudRegions: " + abstractCloudRegions.get(0).getMask());
+        System.out.println("clouds: " + clouds.get(0).getMask());
+
+
+        assertThat(abstractCloudRegions).isEqualTo(clouds);
+        assertThat(abstractCloudRegions.get(0).getId()).isEqualTo(clouds.get(0).getId());
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER_ROLE)
+    public void shouldReturnEmptyListOfCloudRegionsWithoutPermission() {
+        initAclEntity(region);
+        doReturn(clouds).when(cloudRegionManager).loadAll();
+
+        cloudRegionApiService.loadAll();
+
+        assertThat(clouds).isEmpty();
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnCloudRegionForAdmin() {
+        doReturn(region).when(cloudRegionManager).load(region.getId());
+
+        AbstractCloudRegion load = cloudRegionApiService.load(region.getId());
+
+        assertThat(load).isEqualTo(region);
+        assertThat(load.getId()).isEqualTo(clouds.get(0).getId());
```

Comment:
Why do we need such checks?

Reply:
It seemed to me that an additional check would not be superfluous, but perhaps you are right. I will remove it.